tests/unit/test_flist.py

```diff
@@ -82,30 +87,61 @@ def test_api_foldLeft(self):
     self.assertEqual(120.0, alist.foldLeft(1.0, lambda a, e: a * e))
 
   def test_api_foldLeft_with_short_stop(self):
+    """Testing API Cons.foldLeft with arg short_stop_func."""
     alist = fl(1, 2, 3, 4, 5)
-
-    result = alist.foldLeft(
-        0, lambda a, b: a + b,
-        short_stop_func=lambda v: v > 3)
+    f = lambda a, b: a + b # noqa
+    # test corner case where the func stops at identity
+    result = alist.foldLeft(0, f, short_stop_func=lambda v: v > -1)
+    self.assertIsInstance(result, Nil)
 
     # The folding operation will stop at element 4.
     # So we have folding result as 0 + 1 + 2 + 3 = 6
+    result = alist.foldLeft(0, f, short_stop_func=lambda v: v > 3)
     self.assertEqual(result, 6)
 
+  def test_api_foldReft(self):
+    """Testing API Cons.foldRight."""
+    alist = fl(1, 2, 3, 4, 5)
+    self.assertEqual('123450', alist.foldRight(0, lambda a, e: "{}{}".format(a, e)))
+    self.assertEqual(15, alist.foldRight(0, lambda a, e: a + e))
+    self.assertEqual(120.0, alist.foldLeft(1.0, lambda a, e: a * e))
+
+  def test_api_foldRight_with_short_stop(self):
+    """Testing API Cons.foldRight with arg short_stop_func."""
+    alist = fl(1, 2, 3, 4, 5)
+    f = lambda a, b: a + b # noqa
+    # test corner case where the func stops at identity
+    result = alist.foldRight(0, f, short_stop_func=lambda v: v > -1)
+    self.assertIsInstance(result, Nil)
+
+    # The folding operation will stop at element 4.
+    # So we have folding result as 3 + 4 + 5 = 15
+    result = alist.foldRight(0, f, short_stop_func=lambda v: v > 3)
+    self.assertEqual(result, 15)
+
   def test_gapi_fl(self):
     """Testing global API:fl to create object of List."""
-    alist = fl(1, 2, 3)
-    self.assertEqual('[1, 2, 3, NIL]', str(alist))
-    self.assertEqual(3, alist.size())
-    self.assertEqual(3, alist.length())
-    self.assertEqual(1, alist.head())
-    self.assertEqual('[2, 3, NIL]', str(alist.tail()))
+    # case input element (int/float/str) + list + Generator
+    def firstevens(n: int):
+      num = 0
+      while num <= n:
+        yield num
+        num += 2
+    alist = fl(-1, ['a', 'b', 'c'], -10.2, firstevens(6), 'Hi')
+    self.assertEqual('[-1, c, b, a, -10.2, 6, 4, 2, 0, Hi, NIL]', str(alist))
+    self.assertEqual(10, alist.size())
+    self.assertEqual(10, alist.length())
+    self.assertEqual(-1, alist.head())
+    self.assertEqual('[c, b, a, -10.2, 6, 4, 2, 0, Hi, NIL]', str(alist.tail()))
 
   def test_gapi_comp(self):
     """Testing global API:comp to generate composition from given list."""
+    # corner case
+    fpu_comp_list = comp([], random.randint(0, 10))
+    self.assertIsInstance(fpu_comp_list, Nil)
+
     # Composition for 2 element of [1, 2, 3] will be [(1, 2), (1, 3), (2, 3)]
     fpu_comp_list = comp([1, 2, 3], 2)
-
     self.assertEqual('[(1, 2), (1, 3), (2, 3), NIL]', str(fpu_comp_list))
```

Comment:
Refer to [`Parametrizing tests`](https://docs.pytest.org/en/7.1.x/example/parametrize.html). Thanks for adding more unit test case to increase coverage. Good job.